design-spec/priviledge-decentralized-swupdates.tex

```diff
@@ -177,32 +177,483 @@ \section{Design}
 % architecture we assume: for instance, we assume that we have a ledger
 % components that will execute the update protocol somehow.
 
+% Updates are activated.
+
+% Maybe say that we assume the blockchain time is split into epochs.
+
 \subsection{Update types}
 \label{sec:update-types}
 
 % What is an update? How are updates related to the blockchain evolution and/or
 % how are they related to governance.
-
-% What aspects of a blockchain are we updating?
+Updates are the means by which the blockchain system and related ecosystem (e.g.
+wallets, explorers) can evolve. We distinguish the following update types:
+\begin{description}
+\item[Protocol] These updates affect the consensus rules by which nodes agree on
+  a common blockchain. Protocol updates fall in one of the following two
+  categories:
```

Comment:
I think the main distinction is
- Consensus affecting : They require some synchronization mechanism (one of which is signaling) so that a chain split is avoided. These can be further distinguished to static updates (required installation of new software) and dynamic updates (dont require installation of new software but affect consensus e.g., max block size). Note that in the latter case the new parameter values will take effect in a synchronized manner automatically by the update mechanism (e.g., in the beginning of the next epoch) and thus there is no need for a manual synchronization through signaling - the necessary synchronization is embedded in the update mechanism.
- Consensus neutral: They do not affect the consensus and thus can take effect independently for each participant. However, for updates that require a significant downtime period, caution must be taken so that not all parties go off-line simultaneously, since that might affect the security assumptions of the blockchain (e.g., honest majority might not hold for a period of time). These updates can also be distinguished to static and dynamic ones (e.g. parameter updates that dont affect consensus e.g., number of SPOs).